homeassistant/components/unifiprotect/services.py

```diff
@@ -165,6 +177,16 @@ def remove_zone() -> None:
     await camera.queue_update(remove_zone)
 
 
+async def ptz_camera_goto_preset(hass: HomeAssistant, call: ServiceCall) -> None:
+    """Go to preset for PTZ camera."""
+    slot: int = call.data[ATTR_PTZ_SLOT]
+    camera = _async_get_ufp_camera(hass, call)
+    try:
+        await camera.goto_ptz_slot(slot=slot)
```

Comment:
Can we get a list of the slots and maybe names and implement as a `select` entity instead?

Reply:
I have tried that first, but couldn't get the async call to work without a much deeper change in how the entities are getting setup. 
another problem is that, as far as I can tell, I cannot get the current preset that the camera is on from protect. which means that I can't update the entity with current position on boot or when a change happens outside of homeassistant.

Both of these issues pushed me to implement an action that calls the preset id.

Reply:
I think we might be able to call `get_ptz_presets`

https://github.com/uilibs/uiprotect/blob/624101584c2c7460359635ca9fc6d21bca3e863a/src/uiprotect/data/types.py#L705
```
class PTZPreset(BaseModel):
    id: str
    name: str
    slot: int
    ptz: PTZPresetPosition
```

That provides a list of names for the presets

https://github.com/uilibs/uiprotect/blob/624101584c2c7460359635ca9fc6d21bca3e863a/src/uiprotect/data/types.py#L694
You might also be able to call `get_ptz_position` and than iterate the ptz_presets to see if its set to a pre-existing slot

Reply:
ok I will try to resolve the issue I had when called get_ptz_presets when setting up the entity. lets close this PR for now.